backend/entityservice/tasks/solver.py

```diff
@@ -15,23 +15,14 @@ def solver_task(similarity_scores_filename, project_id, run_id, lenf1, lenf2, pa
     solver_task.span.log_kv({'lenf1': lenf1, 'lenf2': lenf2, 'filename': similarity_scores_filename})
     score_file = mc.get_object(config.MINIO_BUCKET, similarity_scores_filename)
     log.debug("Creating python sparse matrix from bytes data")
-    similarities = similarity_matrix_from_csv_bytes(score_file.data)
+    candidate_pairs = anonlink.serialization.load_candidate_pairs(score_file)
     log.info("Calculating the optimal mapping from similarity matrix")
 
-    groups = anonlink.solving.greedy_solve(similarities)
+    groups = anonlink.solving.greedy_solve(candidate_pairs)
 
     log.debug("Converting groups to mapping")
-    mapping = {}
-    for group in groups:
-        assert len(group) == 2  # We only have 2 datasets.
-        i0, i1 = group
-        if i0 > i1:
-            i0, i1 = i1, i0
-        dset_i0, rec_i0 = i0
-        assert dset_i0 == 0
-        dset_i1, rec_i1 = i1
-        assert dset_i1 == 1
-        mapping[str(rec_i0)] = str(rec_i1)  # Strings make JSON happy.
+    mapping = {str(i): str(j)
+               for i, j in anonlink.solving.pairs_from_groups(groups)}
```

Comment:
Perhaps we can get rid of the cast to `str` - are the `candidate_pairs` loaded via `anonlink.serialization.load_candidate_pairs(score_file)` strings already?

Reply:
They are integers when they are loaded by `load_candidate_strings`. `greedy_solve` requires an array of integers and it will not accept strings.